src/java/org/apache/cassandra/index/sai/disk/v2/sortedterms/SortedTermsReader.java

```diff
@@ -104,13 +105,12 @@ public SortedTermsReader(@Nonnull FileHandle termsData,
     }
 
     /**
-     * Returns the point id (ordinal) of the target term or the next greater if no exact match found.
-     * If reached the end of the terms file, returns <code>Long.MAX_VALUE</code>.
-     * Complexity of this operation is O(log n).
-     *
-     * @param term target term to lookup
+     * Returns the point id (ordinal) associated with the greatest term less than or equal to the given term, or
+     * <code>Long.MAX_VALUE</code> if there is no such term.
+     * @param term
+     * @return
      */
-    public long getPointId(@Nonnull ByteComparable term)
+    public long ceiling(@Nonnull ByteComparable term)
```

Comment:
@jbellis - I just want to spend a little more time discussing the naming before coming to a decision. After looking through the code base, ceiling and floor are used in the exact way we want for this PR. Notably, the `PartitionIndex`, which also extends the trie `Walker`  class, uses `ceiling` and `floor` to get the respective results.

https://github.com/datastax/cassandra/blob/1f9fc75f80796cddaad039d2d51ac1c384cefbb7/src/java/org/apache/cassandra/io/sstable/format/trieindex/PartitionIndex.java#L288-L335

What is your preference?

Reply:
As an aside, that implementation for ceiling and floor looks like exactly what we want for this class. `TrieRangeIterator` seems to be more generic, which comes at a cost of keeping a stack for each traversed trie node.